src/client/mod.rs

```diff
@@ -351,6 +371,7 @@ impl IntoFuture for ClientBuilder {
                 #[cfg(feature = "voice")]
                 voice_manager: voice_manager.as_ref().map(Arc::clone),
                 ws_url: Arc::clone(&ws_url),
+                ws_proxy,
```

Comment:
Can this be wrapped in `Arc` like `ws_url`, and preprocessed into `Url` to fail here if the user provides an invalid URL.

Reply:
Yes it can. Finished

Reply:
`ws_url` is only parsed into a `Url` inside of the `gateway::shard::connect` helper function. I think for consistency's sake the same should be done with `ws_proxy` - a followup PR could move the parsing logic for both to be earlier in the initialization stage.